File: src/algorithms/is-circular-sentence.js
/**
 * Given a string sentence, returns true if it is circular, false otherwise.
 * 
 * A sentence is circular if:
 *     • The last character of a word is equal to the first character of the
 *       next word.
 *     • The last character of the last word is equal to the first character
 *       of the first word.
 * 
 * @param {string} sentence - A string of lowercase English letters 
 * @returns {boolean} - Returns true if sentence is circular
 */
function isCircularSentence (sentence) {
  let words = sentence.split(' ');
  for (let i = 0; i < words.length - 1; i++) {
    let len = words[i].length - 1;
    if (words[i][len] !== words[i + 1][0]) {
      return false;
    }
  }
  if (sentence[0] === sentence[sentence.length - 1]) {
    return true;
  }
  return false;
}

File: src/algorithms/is-circular-sentence.py
def is_circular_sentence(sentence):
    """A sentence is a list of words that are separated by a single space with 
    no leading or trailing spaces. For example, "Hello World", "HELLO", "hello 
    world hello world" are all sentences. Words consist of only uppercase and 
    lowercase English letters. Uppercase and lowercase English letters are 
    considered different. A sentence is circular if:
    • The last character of a word is equal to the first character of the next 
    word.
    • The last character of the last word is equal to the first character of 
    the first word.
    For example, "leetcode exercises sound delightful", "eetcode", "leetcode 
    eats soul" are all circular sentences. However, "Leetcode is cool", "happy 
    Leetcode", "Leetcode" and "I like Leetcode" are not circular sentences.
    Given a string sentence, return true if it is circular. Otherwise, return 
    false.

    Args:
        sentence (str): A string of words consisting of English letters and 
        spaces.

    Returns:
        Boolean: Returns true if sentence is circular.
    """
    words = sentence.split()
    for i in range(len(words) - 1):
        if words[i][-1] != words[i + 1][0]:
            return False
    if sentence[0][0] == sentence[len(sentence) - 1][-1]:
        return True
    return False
